PowerShell/Edit-PSFileContentTypeCase.ps1

```diff
@@ -0,0 +1,134 @@
+<#
+.SYNOPSIS
+Updates PowerShell files so their types use the correct case.
+
+.DESCRIPTION
+The `Edit-TypeCase.ps1` script searches a list of paths for PowerShell files (*.ps1, *.psm1) and updates them to use the correct case for their types. The WebMD Health Services standard is that all types should use the case of the underlying type, unless it is a type acclerator that is an alias to a type with a different name (e.g. `int` for `Int32`, `switch` for `SwitchParameter`, etc.).
+
+.EXAMPLE
+Edit-PSFileContentTypeCase.ps1 -Path . -WhatIf
+
+Demonstrates how to call this script and preview the changes that will be made.
+
+.EXAMPLE
+Edit-PSFileContentTypeCase.ps1 -Path . 
+
+Demonstrates how to call this script to make changers to your files.
+#>
+[CmdletBinding(SupportsShouldProcess)]
+param(
+    [Parameter(Mandatory)]
+    # The paths to search.
+    [String[]]$Path
+)
+
+#Requires -Version 5.1
+Set-StrictMode -Version 'Latest'
+
+function Find-Type
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory)]
+        [String]$Name
+    )
+
+    if( $typeNames.Count -eq 0 )
+    {
+        $typeNames['ordered'] = 'ordered'
+
+        [PSObject].Assembly.GetType('System.Management.Automation.TypeAccelerators')::Get.GetEnumerator() |
+            ForEach-Object { [pscustomobject]@{ Name = $_.Key ; Type = $_.Value } } |
+            Where-Object { $_.Type.Name -notlike $_.Name -or $_.Type.Namespace -ne 'System' } |
+            Where-Object { $_.Type.Name -notlike '*Attribute' } |
+            ForEach-Object { $typeNames[$_.Name] = $_.Name.ToLowerInvariant() }
+
+        foreach( $assembly in [AppDomain]::CurrentDomain.GetAssemblies())
+        {
+            try
+            {
+                $types = $assembly.GetExportedTypes()
+            }
+            catch
+            {
+                continue
+            }
+
+            foreach( $type in $types )
+            {
+                if( $type.Namespace -ne 'System' )
+                {
+                    continue
+                }
+                $typeNames[$type.Name] = $type.Name
+            }
+        }
+    }
+
+    if( -not $typeNames.ContainsKey($Name) )
+    {
+        Write-Error -Message ('Type "{0}" not found.' -f $Name)
+        return
+    }
+
+    return $typeNames[$Name]
+}
+
+$typeNames = @{}
+
+$searchRegex = [regex]'\[([A-Za-z]+)(\[\])?\]' 

```

Comment:
Add a zero or more space patterns around the brackets? Catches code that looks like `[ string[ ] ]` ?